README.md

```diff
@@ -293,6 +270,239 @@ See event-component.ts in apitest for a complete example of broadcasting user me
+1. **Embedding via Events**\
+Components are obtained on demand by events. An event handler will construct or otherwise fetch the component and return it. The event handler will also add the component's root HTML element within Golden Layout's own DOM hierarchy sub-tree. This is the binding method introduced in version 2.
+1. **Virtual via Registration**\
+A component's constructor/factory is registered with Golden Layout. Golden Layout will instantiate an instance when required. The component will use the same positioning as virtual components however Golden Layout will handle all the events internally.
+1. **Virtual via Events** (Virtual Components)\
+With virtual components, Golden Layout never interacts directly with components. The application controls the construction/allocation, destruction/deallocation and positioning of components. Golden Layout will advise the application when components are needed and no longer needed via events. It will also advise about components' positioning via events. This allows an application to control the placement of components in the DOM hierarchy and components' root HTML element's ancestors are not reparented when the layout is changed.
+
+#### Embedding via Registration
+
+Registering a component and specifying static positioning is the classic GoldenLayout approach to binding components. The components are registered with GoldenLayout and specify a constructor or callback function used to create a component whenever a new instance is needed in the layout. When the constructor or callback is invoked, it is passed a container object which includes a HTML element. The constructor or callback will create the object and make its top level HTML element a child of the container's HTML element. The component is then part of the Golden Layout's DOM hierarchy. Whenever the layout is re-arranged, the GoldenLayout DOM is adjusted to reflect the new layout hierarchy. Effectively this involves the ancestors of components' root HTML elements being reparented when a layout is changed.
+
+The following functions can be used to register components.
+* `GoldenLayout.registerComponent()`
+* `GoldenLayout.registerComponentConstructor()`
+* `GoldenLayout.registerComponentFactoryFunction()`
+* `GoldenLayout.registerComponentFunction()`
+* `GoldenLayout.registerGetComponentConstructorCallback()`
+
+#### Embedding via Events
+
+To give applications more control over the allocation of components, you can bind components with events instead of registration. If a handler is assigned to the event `VirtualComponent.bindComponentEvent` it will be fired whenever a new component is needed. The handler should:
```

Comment:
VirtualComponent.bindComponentEvent seems wrong here, shouldn't that be GoldenLayout.bindComponentEvent?

Reply:
Should be VirtualLayout.bindComponentEvent. Fixed. If you only use event binding (and no registration classes, you can instantiate VirtualLayout instead of GoldenLayout. Avoids overhead of registration functions. VirtualLayout now has all the logic that GoldenLayout previously had so can be used standalone.